plugins/miopen_legacy_plugin/integration_tests/batchnorm_bwd_integration_test.cpp

```diff
@@ -0,0 +1,393 @@
+// Copyright © Advanced Micro Devices, Inc., or its affiliates.
+// SPDX-License-Identifier:  MIT
+
+#include <cmath>
+#include <gtest/gtest.h>
+#include <hip/hip_runtime.h>
+#include <memory>
+#include <random>
+#include <vector>
+
+#include <hipdnn_frontend/attributes/tensor_attributes.hpp>
+#include <hipdnn_frontend/graph.hpp>
+#include <hipdnn_frontend/utilities.hpp>
+#include <hipdnn_sdk/test_utilities/cpu_fp_reference_implementation.hpp>
+#include <hipdnn_sdk/test_utilities/cpu_fp_reference_validation.hpp>
+#include <hipdnn_sdk/test_utilities/test_utilities.hpp>
+#include <hipdnn_sdk/utilities/migratable_memory.hpp>
+#include <hipdnn_sdk/utilities/tensor.hpp>
+
+using namespace hipdnn_frontend;
+using namespace hipdnn_frontend::graph;
+using namespace hipdnn_sdk::utilities;
+using namespace hipdnn_sdk::reference_test_utilities;
+
+namespace
+{
+
+struct Bn_2d_test_case
+{
+    int64_t n;
+    int64_t c;
+    int64_t h;
+    int64_t w;
+
+    friend std::ostream& operator<<(std::ostream& ss, const Bn_2d_test_case& tc)
+    {
+        return ss << "(n:" << tc.n << " c:" << tc.c << " h:" << tc.h << " w:" << tc.w << ")";
+    }
+
+    std::vector<int64_t> get_dims() const
+    {
+        return {n, c, h, w};
+    }
+};
+
+template <typename Input_type, typename Intermediate_type>
+struct Batchnorm_2d_tensor_bundle
+{
+    Batchnorm_2d_tensor_bundle(const std::vector<int64_t>& dims,
+                               unsigned int seed = 1,
+                               const Tensor_layout& layout = Tensor_layout::NCHW)
+        : derived_dims({1, dims[1], 1, 1})
+        , x_tensor(dims, layout)
+        , dy_tensor(dims, layout)
+        , dx_tensor(dims, layout)
+        , scale_tensor(derived_dims)
+        , dscale_tensor(derived_dims)
+        , dbias_tensor(derived_dims)
+        , mean_tensor(derived_dims)
+        , inv_variance_tensor(derived_dims)
+    {
+        x_tensor.fill_with_random_values(
```

Comment:
We could also have something similar to https://github.com/ROCm/composable_kernel/blob/develop/include/ck_tile/host/fill.hpp